File: Counter/Counter_exercise_Frequency Problem.py
def same_frequency(str1, str2):
    """
    輸入兩個字串，確認其字母出現的頻率是否相同
    """
    counter1 = {}
    counter2 = {}
    for i in range(len(str1)):
        if str1[i] not in counter1.keys():
            counter1[str1[i]] = 1
        elif str1[i] in counter1.keys():
            counter1[str1[i]] += 1
    for i in range(len(str2)):
        if str2[i] not in counter2.keys():
            counter2[str2[i]] = 1
        elif str2[i] in counter2.keys():
            counter2[str2[i]] += 1
    if counter1 == counter2:
        return True
    elif counter1 != counter2:
        return False

a = same_frequency("abab", "aabb")
b = same_frequency("aaab", "abab")
c = same_frequency("abbb", "bbba")
d = same_frequency("abcd", "abcdefg")
print(a, b, c, d)

File: Counter/Counter_exercise_Frequency Problem.js
function sameFrequency(str1, str2) {
  // 輸入兩個字串，確認其字母出現的頻率是否相同
  let arr1 = str1.split("");
  let arr2 = str2.split("");
  if (arr1.length != arr2.length) {
    return false;
  }
  let counter1 = {};
  let counter2 = {};
  for (let i = 0; i < arr1.length; i += 1) {
    if (counter1[arr1[i]]) {
      counter1[arr1[i]] += 1;
    } else {
      counter1[arr1[i]] = 1;
    }
  }
  for (let i = 0; i < arr2.length; i += 1) {
    if (counter2[arr2[i]]) {
      counter2[arr2[i]] += 1;
    } else {
      counter2[arr2[i]] = 1;
    }
  }
  for (let property in counter1) {
    if (!counter2[property]) {
      return false;
    }
    if (counter2[property] !== counter1[property]) {
      return false;
    }
  }
  return true;
}

let a = sameFrequency("abab", "aabb");
let b = sameFrequency("aaab", "abab");
let c = sameFrequency("abbb", "bbba");
let d = sameFrequency("abcd", "abcdefg");
console.log(a, b, c, d);
